petclinic/actors/clinicapi/src/lib.rs

```diff
@@ -225,9 +247,38 @@ async fn get_owner(ctx: &Context, owner_id: &str) -> RpcResult<HttpResponse> {
     }
 }
 
+fn get_pet_type(pts: &[PetType], id: u64) -> AugmentedPetType {
+    pts.iter().find(|p| p.id == id).map_or_else(
+        || AugmentedPetType {
+            id: 0,
+            name: "Unknown".to_string(),
+        },
+        |p| AugmentedPetType {
+            id: id,
+            name: p.name.to_string(),
+        },
+    )
+}
+
 fn deser<'de, T: Deserialize<'de>>(raw: &'de [u8]) -> RpcResult<T> {
     match serde_json::from_slice(raw) {
         Ok(t) => t,
         Err(e) => Err(RpcError::Deser(format!("{}", e))),
     }
 }
+
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+struct AugmentedPet {
+    pub birthdate: Date,
+    pub id: u64,
+    #[serde(default)]
+    pub name: String,
+    #[serde(rename = "petType")]
+    pub pet_type: AugmentedPetType,
+}
```

Comment:
I'm just imagining a robo-pet when I see `AugmentedPet`

Reply:
:robot: :dog:

Reply:
That's a _very different_ kind of pet clinic then :grin: